src/app/app-interactive-map/app-interactive-map.component.ts

```diff
@@ -127,15 +93,37 @@ export class AppInteractiveMapComponent implements OnInit, AfterViewInit {
 
   ngAfterViewInit(): void {
     const element = d3Select(this.elRef.nativeElement.querySelector('.escher-builder'));
-    this.builderSubject.next(
-      escher.Builder(
-        null,
-        null,
-        null,
-        element,
-        this.escherSettings,
-      ),
+    this.builder = escher.Builder(
+      null,
+      null,
+      null,
+      element,
+      this.escherSettings,
     );
+    this.store.pipe(
+      selectNotNull((store) => store.interactiveMap.mapData),
+    ).subscribe((map) => {
+      setTimeout(() => this.builder.load_map(map), 0);
```

Comment:
This can be a little bit hacky, but what is happening is exactly the same that is happening with the dialogs, and is this:
    Initializing escher takes time, then while escher is initializing we are trying to load the map, when escher is initialized it is initialized with the map to null, and as we tried to load the map before it finished initializing the map that we have loaded remains in the forget and the map loads the default value, null.